faiss/impl/ProductAdditiveQuantizer.h

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include <faiss/Index.h>
+#include <faiss/impl/AdditiveQuantizer.h>
+#include <faiss/impl/LocalSearchQuantizer.h>
+#include <faiss/impl/ResidualQuantizer.h>
+
+namespace faiss {
+
+/** Product Additive Quantizers
+ * TODO: add some docs
+ */
+struct ProductAdditiveQuantizer : AdditiveQuantizer {
+    size_t nsplits; ///< number of sub-vectors we split a vector into
+
+    std::vector<AdditiveQuantizer*> quantizers;
+
+    ProductAdditiveQuantizer(
+            size_t d,
+            const std::vector<AdditiveQuantizer*>& aqs,
```

Comment:
please clarify ownership of the AddtitiveQuantizer's. Is it transferred to the ProductAdditiveQuantizer?

Reply:
The ownership is not transferred to the ProductAdditiveQuantizer. I will clarify it in the docs.